tests/components/keenetic_ndms2/test_config_flow.py

```diff
@@ -0,0 +1,129 @@
+"""Test Keenetic NDMS2 setup process."""
+
+from ndms2_client import ConnectionException
+from ndms2_client.client import RouterInfo
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components import keenetic_ndms2 as keenetic
+from homeassistant.const import CONF_HOST, CONF_NAME
+
+from . import MOCK_DATA, MOCK_OPTIONS
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(name="connect")
+def mock_keenetic_connect():
+    """Mock connection routine."""
+    with patch("ndms2_client.client.Client.get_router_info") as mock_get_router_info:
+        mock_get_router_info.return_value = RouterInfo(
+            name="Mock Router",
+            fw_version="3.0.4",
+            fw_channel="stable",
+            model="mock",
+            hw_version="0000",
+            manufacturer="pytest",
+            vendor="foxel",
+            region="RU",
+        )
+        yield
+
+
+@pytest.fixture(name="connect_error")
+def mock_keenetic_connect_failed():
+    """Mock connection routine."""
+    with patch(
+        "ndms2_client.client.Client.get_router_info",
+        side_effect=ConnectionException("Mocked failure"),
+    ):
+        yield
+
+
+async def test_flow_works(hass, connect):
+    """Test config flow."""
+
+    result = await hass.config_entries.flow.async_init(
+        keenetic.DOMAIN, context={"source": "user"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
Please patch the integration setup functions on all create entry results to avoid setting up the whole integration. See the scaffold script test template for example.